.github/workflows/build_commits.yaml

```diff
@@ -0,0 +1,56 @@
+name: build_commits
+
+on: [pull_request]
+
+jobs:
+  build_commits:
+    name: Check if build works for every commit
+    runs-on: ubuntu-latest
```

Comment:
```suggestion
  build_commits:
    name: Check if build works for every commit
    runs-on: ubuntu-latest
    timeout-minutes: 60
```
You know why 😁

Reply:
What happens if this timeout is hit? Do we just fail? That means we can't make this action required to merge PRs as they may simply have too many commits. I think that's probably fine (we can improve later if needed) but wanted to raise it.

Reply:
After the timeout Github automatically cancels the job. Yes we can surely change it once it's confirmed that we are on the right track and the commits are building fine.